pints/_mcmc/_dual_averaging.py

```diff
@@ -0,0 +1,241 @@
+#
+# Dual Averaging step size and mass matrix adaption method for NUTS and HMC
+# samplers
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+
+
+class DualAveragingAdaption:
+    """
+    Implements a Dual Averaging scheme to adapt the step size ``epsilon``, as
+    per [1]_, and estimates the (fully dense) inverse mass matrix using the
```

Comment:
Is the dual averaging from section 3.2.1 in [1]? Or, alternatively, from one of the algorithms? In either case, can we point to it exactly as this is a long paper...

Reply:
yes, section 3.2.1 and algorithm 6. I've added this to the docstring